src/modelbench/run.py

```diff
@@ -190,18 +191,27 @@ def consistency_check(journal_path, verbose):
 
 
 def find_suts_for_sut_argument(sut_uids: List[str]):
+    # TODO: Put object initialization code in once place shared with modelgauge.
+    # Make sure we have all the secrets we need.
+    secrets = load_secrets_from_config()
+    missing_secrets = []
+    unknown_uids = []
     suts = []
-    registered_sut_keys = set(i[0] for i in SUTS.items())
     for sut_uid in sut_uids:
-        if sut_uid in registered_sut_keys:
-            suts.append(ModelGaugeSut.for_key(sut_uid))
-        else:
-            valid_suts = sorted(registered_sut_keys, key=lambda x: x.lower())
-            valid_suts_str = "\n\t".join(valid_suts)
-            raise click.BadParameter(
-                f"Unknown uid '{sut_uid}'. Valid options are: {valid_suts_str}",
-                param_hint="sut",
-            )
+        try:
+            missing_secrets.extend(SUTS.get_missing_dependencies(sut_uid, secrets=secrets))
+            suts.append(SUTS.make_instance(sut_uid, secrets=secrets))
```

Comment:
A thought, not for this PR: while I'd normally be in favor of the client code injecting dependencies into the objects it creates (`secrets` in this case) because it makes testing easier, in this case it feels like the SUT should know how to configure itself, and the client code shouldn't need to. WDYT?

Reply:
Dependency injection allows us to know all of the required secrets at the beginning of a run + cleanly exit and report missing secrets without having to initialize any objects, which might be expensive. I believe those were the guiding principles that lead us to this paradigm.